feature-utils/silly-i18n/rollup-plugin.cjs

```diff
@@ -0,0 +1,57 @@
+const fs = require("fs");
+const path = require("path");
+const util = require("util");
+const glob = require("glob");
+
+const DIRNAME = "locales";
+const FILEEXT = ".json";
+const PLUGIN = "silly-i18n";
+const SUFFIX = `?${PLUGIN}`;
+
+exports.default = function (options) {
+    return {
+        name: PLUGIN,
+
+        async resolveId(source, importer) {
+            if (source == `!${PLUGIN}`) {
+                for (let dir = importer; dir != (dir = path.dirname(dir)); ) {
```

Comment:
The second part of the loop is a bit confusing. Do you need to reassign `dir`?

Reply:
Yes, I do. The new value is used in the loop's body. Also we have to check here that the previous value is different in order to detect that we reached the top level.

This code might be somewhat less confusing if using a `while (true)` loop but that isn't permitted by the ESLint configuration.

Reply:
A while loop would have been a bit less confusing, I guess. Initial dir is assigned before the while loop, it will run while the current dir is different from the previous dir.

Reply:
In the first iteration `dir` is the path to the directory of the module that performs the import, in the second iteration it's the next director up, and so on. Note that the expression after the first semicolon in the loop header is performed before each iteration including the first one. I'm not exactly sure what issues if any you try to hint at?